File: Find Kth Bit in Nth Binary String.cpp
class Solution {
public:
    char findKthBit(int n, int k) {
        if(n==1 && k==1)
            return '0';
        int p = pow(2,n);
        int mid = p/2;
        if(k==mid)
            return '1';
        if(k<mid)
            return findKthBit(n-1, k);
        char answer = findKthBit(n-1, p-k);
        if(answer == '0')
            return '1';
        return '0';
    }
};

File: Find Kth Bit in Nth Binary String.py
class Solution:
    def calculate(self, n):
        if n == 1:
            return 1
        return (self.calculate(n-1) * 2 +1)

    def findKthBit(self, n: int, k: int) -> str:
        if k == 1:
            return "0"
        length = self.calculate(n)
        if k <= length//2:
            return self.findKthBit(n-1, k)
        if k == (length//2)+1:
            return "1"
        return str(1 - int(self.findKthBit(n-1,length-k+1)))
        



